corehq/apps/export/views.py

```diff
@@ -1707,37 +1722,95 @@ def _check_export_size(self, export_instances, filters):
+
 
+@location_safe
 class DownloadNewFormExportView(GenericDownloadNewExportMixin, DownloadFormExportView):
     urlname = 'new_export_download_forms'
-    filter_form_class = FilterFormESExportDownloadForm
+    filter_form_class = EmwfFilterFormExport
+    export_filter_class = LocationRestrictedMobileWorkerFilter
 
     def _get_export(self, domain, export_id):
         return FormExportInstance.get(export_id)
 
-    def get_filters(self, filter_form_data):
+    def get_filters(self, filter_form_data, mobile_user_and_group_slugs):
         filter_form = self._get_filter_form(filter_form_data)
-        form_filters = filter_form.get_form_filter()
+        form_filters = filter_form.get_form_filter(mobile_user_and_group_slugs,
+                                                   self.request.can_access_all_locations)
+        if not self.request.can_access_all_locations:
+            form_filters.append(self.scope_filter())
         return form_filters
 
+    def scope_filter(self):
+        # Filter to be applied in AND with filters for export for restricted user
+        # Restricts to forms submitted by users at accessible locations
+        accessible_user_ids = (UserES()
+            .user_ids_at_accessible_locations(
+                self.request.domain, self.request.couch_user
+        ))
+        return FormSubmittedByFilter(accessible_user_ids)
+
 
 class BulkDownloadNewFormExportView(DownloadNewFormExportView):
     urlname = 'new_bulk_download_forms'
     page_title = ugettext_noop("Download Form Exports")
 
 
+@location_safe
 class DownloadNewCaseExportView(GenericDownloadNewExportMixin, DownloadCaseExportView):
     urlname = 'new_export_download_cases'
     filter_form_class = FilterCaseESExportDownloadForm
+    export_filter_class = CaseListFilter
 
     def _get_export(self, domain, export_id):
         return CaseExportInstance.get(export_id)
 
-    def get_filters(self, filter_form_data):
+    def get_filters(self, filter_form_data, mobile_user_and_group_slugs):
         filter_form = self._get_filter_form(filter_form_data)
-        form_filters = filter_form.get_case_filter()
+        form_filters = filter_form.get_case_filter(mobile_user_and_group_slugs,
+                                                   self.request.can_access_all_locations)
+        if not self.request.can_access_all_locations:
+            form_filters.append(self.scope_filter())
         return form_filters
 
+    def scope_filter(self):
+        # Filter to be applied in AND with filters for export to add scope for restricted user
+        # Restricts to cases owned by accessible locations and their respective users Or Cases
+        # Last Modified by accessible users
+        accessible_location_ids = (SQLLocation.active_objects.accessible_location_ids(
+            self.request.domain,
+            self.request.couch_user)
+        )
+        accessible_user_ids = UserES().user_ids_at_locations(accessible_location_ids)
```

Comment:
There is a catch. Actually, I need accessible_location_ids later for OwnerFilter, hence could not fetch just the users from locations.